ERCS/erc-7871.md

```diff
@@ -0,0 +1,139 @@
+---
+eip: 7871
+title: Wallet Signing API
+description: Adds a JSON-RPC method for requesting a signature from a wallet
+author: Lukas Rosario (@lukasrosario), Jake Moxey (@jxom), Cody Crozier (@wcrozier12), Conner Swenberg (@ilikesymmetry)
+discussions-to: https://ethereum-magicians.org/t/new-erc-wallet-signing-api/22718
+status: Draft
+type: Standards Track
+category: Interface
+created: 2024-01-29
+requires: 191, 712, 5792
+---
+
+## Abstract
+
+Defines a new JSON-RPC method which enables apps to ask a wallet to sign [EIP-191](./eip-191.md) messages.
+
+Applications can use this JSON-RPC method to request a signature over any version of `signed_data` as defined by [EIP-191](./eip-191.md). The new JSON-RPC method allows for support of future [EIP-191](./eip-191.md) `signed_data` versions.
+
+The new JSON-RPC method also supports [EIP-5792](./eip-5792.md)-style `capabilities`, and support for signing capabilities can be discovered using `wallet_getCapabilities` as defined in [EIP-5792](./eip-5792.md).
+
+## Motivation
+
+Wallets and developer tools currently support multiple JSON-RPC methods for handling offchain signature requests. This proposal simplifies wallet & tooling implementations by consolidating these requests under a single `wallet_sign` JSON-RPC method. This also leaves room for new [EIP-191](./eip-191.md) `signed_data` versions without needing to introduce a new corresponding JSON-RPC method.
+
+Furthermore, this new `wallet_sign` method introduces new functionalities via [EIP-5792](./eip-5792.md)-style `capabilities`.
```

Comment:
This is the second hint at `capabilities` without any meat behind it. In this section in particular, you want to explain why this is a desirable property.